### PR TITLE
Drop XML support in favor of JSON

### LINKED ISSUE #45: Remove support for NVD CVE XML feeds
It's probably time we start discussing deprecating support for CVE XML feeds. We've been using NVD CVE JSON 1.0 for half year and the benefits from versions ranges are satisfactory.

In the codebase, however, having XML around makes things a log more complex, including manual edits to the auto-generated JSON schema to support the "nvdcommon" package which merges both XML and JSON into a common interface.

As we add more converters (the so called $vendor2nvd) to this codebase, we will need a proper generated JSON schema that is not edited manually, ever.

The only breaking change I can see is for the cpe2cve -feed flag, which should probably be removed as well if we make JSON the one and only supported feed.

### PR DESCRIPTION
**Pull request by andrestc**
This PR drops the support for XML file format for nvd feeds.

 It also unifies the JSON schema used in different pkgs to a single one, distributed under `cvefeed/jsonschema`. This json schema is kept exactly as it is generated (the previous one, used in `nvdjson` was modified to satisfy some of the Interfaces).

The `-feed` flag was kept for backwards compatibility.

Fixes #45.

### CODE REVIEW

**Comment by fiorix** [comment]
Please squash so we can review :)

**Comment by andrestc** [comment]
PTAL @fiorix

**Comment by n1chre** [comment]
Can we also drop vulndb/jsonschema package then, throughout the whole repo? just use the  one in cvefeed or no?

File: cmd/cpe2cve/cpe2cve.go [hunk 1]
```diff
@@ -237,20 +237,14 @@ func main() {
...
-	case "xml":
-		if dict, err = cvefeed.LoadXMLDictionary(flag.Args()...); err == nil {
-			overrides, err = cvefeed.LoadXMLDictionary(cfg.overrides...)
-		}
-	case "json":
-		if dict, err = cvefeed.LoadJSONDictionary(flag.Args()...); err == nil {
-			overrides, err = cvefeed.LoadJSONDictionary(cfg.overrides...)
-		}
-	default:
+	if cfg.feedFormat != "json" {
```
Comment (fiorix) [inline cmd/cpe2cve/cpe2cve.go]: We're gonna have to drop this after we update our pipelines.

File: cmd/cpe2cve/cpe2cve_test.go [hunk 2]
```diff
@@ -97,7 +90,7 @@ func TestProcessInput(t *testing.T) {
 		memProfile:  "",
 		skip:        getSkip([]int{1, 3}),
 	}
-	for cacheID, cache := range []*cvefeed.Cache{cacheXML, cacheJSON} {
+	for cacheID, cache := range []*cvefeed.Cache{cacheJSON} {
```
Comment (fiorix) [inline cmd/cpe2cve/cpe2cve_test.go]: At some point gotta get rid of this too

**Pull request merged**